src/OpenSSL/SSL.py

```diff
@@ -2243,6 +2244,48 @@ def get_peer_cert_chain(self):
             result.append(pycert)
         return result
 
+    def get_verified_chain(self):
+        """
+        Retrieve the verified certificate chain of the peer including the
+        peer's end entity certificate. It must be called after a session has
+        been successfully established. If peer verification was not successful
+        the chain may be incomplete, invalid, or None.
+
+        :return: A list of X509 instances giving the peer's verified
+                 certificate chain, or None if it does not have one.
+
+        .. versionadded:: 20.0
+        """
+        if hasattr(_lib, 'SSL_get0_verified_chain'):
+            # OpenSSL 1.1+
+            cert_stack = _lib.SSL_get0_verified_chain(self._ssl)
+            if cert_stack == _ffi.NULL:
+                return None
+
+            result = []
+            for i in range(_lib.sk_X509_num(cert_stack)):
+                # TODO could incref instead of dup here
+                cert = _lib.X509_dup(_lib.sk_X509_value(cert_stack, i))
```

Comment:
Yeah let's `X509_up_ref` this instead of duping. Be sure to assert on the result for any call you make (X509_up_ref returns 1 on 1.1.0+ and the refcount in 1.0.2 I believe so it should probably be `_openssl_assert(res >= 1)`.